x/contracts/events/publisher/http.go

```diff
@@ -76,29 +75,43 @@ func (h *httpPublisher) Start(ctx context.Context, contracts ...common.Address)
 				}
 
 				if blockNumber > lastBlock {
-					q := ethereum.FilterQuery{
-						FromBlock: big.NewInt(int64(lastBlock + 1)),
-						ToBlock:   big.NewInt(int64(blockNumber)),
-						Addresses: contracts,
-					}
-
-					logs, err := h.evmClient.FilterLogs(ctx, q)
-					if err != nil {
-						h.logger.Warn("failed to filter logs", "error", err)
-						continue
-					}
-
-					for _, logMsg := range logs {
-						// process log
-						h.subscriber.PublishLogEvent(ctx, logMsg)
+					startBlock := lastBlock + 1
+					success := true
+
+					for startBlock <= blockNumber {
+						endBlock := startBlock + 5000 - 1
+						if endBlock > blockNumber {
+							endBlock = blockNumber
+						}
+
+						q := ethereum.FilterQuery{
+							FromBlock: big.NewInt(int64(startBlock)),
+							ToBlock:   big.NewInt(int64(endBlock)),
+							Addresses: contracts,
```

Comment:
Where are we adding to this dynamically?

Reply:
Oh sorry those changes were still local, pushing them now.